crates/fluvio-protocol/src/core/bytebuf.rs

```diff
@@ -0,0 +1,221 @@
+use std::io::Error;
+use std::io::ErrorKind;
+use std::ops::Deref;
+
+use bytes::{Buf, Bytes, BufMut};
+
+use crate::{Encoder, Decoder, Version};
+
+/// Represnts a SmartModule WASM File bytes.
+///
+/// Provides a `Encoder` and `Decoder` implementation optimized for WASM files
```

Comment:
remove reference to WASM since this optimization is universal

Reply:
Addressed on: [398e679](https://github.com/infinyon/fluvio/pull/2738/commits/398e6792a1ebc8f2e893d574b7bff2d556520a74)